src/main/java/seedu/toluist/dispatcher/CommandDispatcher.java

```diff
@@ -1,46 +1,106 @@
 package seedu.toluist.dispatcher;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
 
 import seedu.toluist.commons.core.EventsCenter;
+import seedu.toluist.commons.core.LogsCenter;
 import seedu.toluist.commons.events.ui.NewResultAvailableEvent;
+import seedu.toluist.controller.AliasController;
 import seedu.toluist.controller.Controller;
 import seedu.toluist.controller.ExitController;
 import seedu.toluist.controller.ListController;
 import seedu.toluist.controller.RedoController;
 import seedu.toluist.controller.StoreController;
 import seedu.toluist.controller.TaskController;
+import seedu.toluist.controller.UnaliasController;
 import seedu.toluist.controller.UndoController;
+import seedu.toluist.controller.ViewAliasController;
+import seedu.toluist.model.CommandAliasConfig;
 import seedu.toluist.ui.Ui;
 
 public class CommandDispatcher extends Dispatcher {
-
+    private final Logger logger = LogsCenter.getLogger(getClass());
     private final EventsCenter eventsCenter = EventsCenter.getInstance();
+    private final CommandAliasConfig aliasConfig = CommandAliasConfig.getInstance();
+
+    public CommandDispatcher() {
+        super();
+        aliasConfig.setReservedKeywords(getControllerKeywords());
+    }
 
     public void dispatch(Ui renderer, String command) {
-        final Controller controller = getBestFitController(renderer, command);
-        final CommandResult feedbackToUser = controller.execute(command);
+        String deAliasedCommand = aliasConfig.dealias(command);
+        logger.info("De-aliased command to be dispatched: " + deAliasedCommand);
```

Comment:
Unaliasing is to remove the alias from the system. here is convert the commands with aliases into one without :-?